libs/langchain/langchain/callbacks/base.py

```diff
@@ -112,6 +112,13 @@ def on_agent_finish(
     ) -> Any:
         """Run on agent end."""
 
+    def on_retry(
```

Comment:
note that i'm not sure what the args/kwargs should be here, so i just did this broad pattern ....